ldm/invoke/generator/diffusers_pipeline.py

```diff
@@ -377,7 +436,7 @@ def latents_from_embeddings(self, latents: torch.Tensor, num_inference_steps: in
                                 callback: Callable[[PipelineIntermediateState], None] = None
                                 ) -> tuple[torch.Tensor, Optional[AttentionMapSaver]]:
         if timesteps is None:
-            self.scheduler.set_timesteps(num_inference_steps, device=self.unet.device)
+            self.scheduler.set_timesteps(num_inference_steps, device=self._model_group.device_for(self.unet))
```

Comment:
would it make sense to add a `def get_device(self, submodel)` function that called through to this? and a `@property def device(self)` that returned `self.get_device(self.unet)` ? just feels like implementation details are leaking here.

Reply:
There already such a `device` on the pipeline class.

I'm ambivalent about adding `get_device(submodel)` to the pipeline. Because I see what you're getting at, but also I think that making use of implementation details of `self` is permissible.

On the other hand, if you've got something in mind that would split these methods off of the class that holds the model group, then making that part of the interface will be important.

Reply:
> making use of implementation details of self is permissible.

yeah, i see that too. i don't have strong feelings and don't mean for this to be a blocker for getting this in. i'm just thinking that down the road it's going to be easy for somebody who isn't being cautious to just type `self.unet.device` without being aware that `_model_group` is a thing they should be paying attention to - and this could lead to hard-to-trace memory issues in the future.